### PR TITLE
Gitignore `.DS_Store` and similar files

### PR DESCRIPTION
**Pull request by mattstein**
## The Problem/Issue/Bug:

One might easily and accidentally commit those annoying macOS `.DS_Store` files to their commit, adding cruft to the project.

## How this PR Solves The Problem:

This uses the `.gitignore` file to prevent those from being noticed or committed by Git.

## Manual Testing Instructions:

🤔 Stage this change, then purposefully try to commit a file named `.DS_Store`. It’ll be trickier, at least.

## Automated Testing Overview:

n/a

## Related Issue Link(s):

Faintly related to #4188; came up during #4336.

## Release/Deployment notes:

n/a

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4337"></a>

### CODE REVIEW

**Review by rfay — APPROVED** [review]
Nothing wrong with that, but I gitignore stuff like that in global gitignore, I recommend to you as well. 

My global .gitignore:
```
.project
.buildpath
.settings
nbproject/
.idea
.DS_Store
db/db_*
docroot/core/phpunit.xml
docroot/drush
docroot/sites/simpletest
docroot/modules/contrib/devel
sites/default/files
*.orig
*.rej
.tarballs
```

**Comment by mattstein** [comment]
Oh, that makes more sense! Thank you.

**Comment by tyler36** [comment]
Yes, a global `.gitignore` is definately great.

However, I think for a project like DDEV that recieves lots of contributions, it's helpful to have a project level one too.

**Comment by rfay** [comment]
May want to add a couple more of those if we're doing this one.

**Comment by mattstein** [comment]
I mean now that we’ve trained me to use a global file, I won’t be noticing whatever those other things are! 😅 🚀

**Comment by rfay** [comment]
Another pro tip: Always use `git add -u` when choosing changes for a commit. Then look and see what might be left over with `git diff`. The leftover stuff is normally cruft, unless you explicitly added a file, in which case `git add <file>`.  If you use `git add -u` you'll almost never check in something you shouldn't have.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@
 /dist
 /.ddev
 /artifacts
+*.DS_Store
```
Comment (rfay) [inline .gitignore:13]:
```suggestion
*.DS_Store
nbproject/
.idea
*.orig
*.rej
```

**Review by rfay — CHANGES_REQUESTED** [review]

**Pull request merged**